### PR TITLE
Port AWS smoke tests to Azure

### PR DESCRIPTION
**Pull request by alexsomesan**
Converted some of the existing AWS tests to Azure.
This is meant as a first step in covering Azure with integration testing.

They are muted until the Azure support in master is stabilized.

### CODE REVIEW

File: tests/smoke/azure/vars/azure-ca.tfvars [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+tectonic_azure_create_dns_zone = false
+tectonic_azure_enable_ssh_external = "true"
+tectonic_azure_etcd_storage_account_type = "Premium_LRS"
+tectonic_azure_etcd_vm_size = "Standard_DS2_v2"
+tectonic_azure_master_vm_size = "Standard_DS2_v2"
+tectonic_azure_worker_vm_size = "Standard_DS2_v2"
+tectonic_azure_master_storage_account_type = "Premium_LRS"
+tectonic_azure_worker_storage_account_type = "Premium_LRS"
+tectonic_azure_ssh_key = "/Users/alex/.ssh/id_rsa.pub"
```
Comment (s-urbaniak) [inline tests/smoke/azure/vars/azure-ca.tfvars]: we should use some default jenkins location and override `TF_VAR_tectonic_azure_ssh_key` for local execution.

File: tests/smoke/azure/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Tectonic Smoke Tests on Azure
+
+This README documents how to run Tectonic smoke tests on Azure. This directory has the following configuration:
+```
+Azure
+├── smoke.sh           # Utility script for planning and creating clusters and running tests
+└── vars               # Terraform tfvars files for Azure smoke tests
+    ├── Azure-exp.tfvars # Default Tectonic configuration + experimental features enabled
```
Comment (s-urbaniak) [inline tests/smoke/azure/README.md]: s/Azure/azure, just to be consistent with the other platforms.

File: tests/smoke/azure/vars/azure-ca.tfvars (hunk 1 above)
Comment (alexsomesan) [inline tests/smoke/azure/vars/azure-ca.tfvars]: Good catch! You're right, this needs to come from Jenkins.
This was just a hack to run them locally.

File: tests/smoke/azure/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+## Credentials
+For now, the smoke tests rely on Azure service principal type of credentials being injected into the environment.
+
+Here is an example of how to set variables that carry Azure credentials:
+```
+export ARM_SUBSCRIPTION_ID="1901df3f-af3b-4a10-8874-58c8db1d5c8f"
+export ARM_CLIENT_ID="66e0e96e-b83e-43f1-b1dc-148e7c9b8f36"
+export ARM_CLIENT_SECRET="a5ec54fc-65a6-4a30-a3ff-80afc7213539"
+export ARM_TENANT_ID="7fc8fb88-785b-4194-830d-acd348ea3b7d"
```
Comment (Quentin-M) [inline tests/smoke/azure/README.md]: Are these random creds?

**Comment by alexsomesan** [comment]
Ofcourse :)

**Comment by alexsomesan** [comment]
Updated with a minimal set of tfvars and hooked them up in jenkins.

With the current state of Azure in master, they mostly fail. Therefore I would like to keep the "do-not-merge" label in place for a little more, while I clean up master to a more stable state.

Please hold off merging until WIP labels are gone.

**Comment by mxinden** [comment]
@alexsomesan Can I remove the `do-not-test` label?

**Comment by alexsomesan** [comment]
@mxinden Yeah, let's try that

**Comment by mxinden** [comment]
@alexsomesan Rebased to get rid of `Jenkinsfile` merge conflict.

**Comment by alexsomesan** [comment]
retest this please

**Comment by Quentin-M** [comment]
They are not numbed entirely just yet ;-)

**Comment by mxinden** [comment]
Closing here. I have moved these changes to #1154 (See 4223d42b00c17e47954e0e2a2f43a0de8d643c1e) so we can test them properly.

**Pull request closed without merging**